roles/container_runtime/templates/daemon.json

```diff
@@ -5,10 +5,15 @@
     "disable-legacy-registry": false,
     "exec-opts": ["native.cgroupdriver=systemd"],
     "insecure-registries": {{ l_docker_insecure_registries }},
-{% if openshift_docker_log_driver  %}
-    "log-driver": "{{ openshift_docker_log_driver }}",
-{%- endif %}
-    "log-opts": {{ l_docker_log_options_dict }},
+    {% if openshift_docker_log_driver  %}
+        "log-driver": "{{ openshift_docker_log_driver }}",
+    {%- endif %}
+    {% if 'max-size' in l_docker_log_options_dict | from_json -%}
+        "log-opts": {{ l_docker_log_options_dict }},
+    {% else -%}
+        {% set new_opts = l_docker_log_options_dict | from_json | combine({'max-size': '50m'}) -%}
```

Comment:
Lets introduce a new var and set it to 50m by default

Reply:
Why would `openshift_docker_options` ever have `max-size` in it if `max-size` is a log option. Also in defaults/main.yml `openshift_docker_options` is set to False.
The method above allows the user to override `openshift_docker_log_options`, and if they don't `max-size` is set to 50m.

Reply:
An admin is always going to customize where needed, our job to set sensible defaults which apply generally, for most use cases.  The ``max-size=50m`` was the suggested default, but perhaps you're correct, it's much too specific.  How about a compromise:

* Make ``journald`` the default, and let it do it's thing.
* Leave all these more specific, fine-grain options blank.

Would that be more realistic and sensible?

Reply:
Oops, right, it should be `openshift_docker_log_options`.

I'm not sure why would we want to silently override user settings. Wouldn't it be best to append `max-size=50m` if its not set - and keep user settings (e.g. 100m) if its already set?

Reply:
Good catch, that was our intention.  So like this instead:

``{% set new_opts = {'max-size': '50m'} | combine(l_docker_log_options_dict | from_json) -%}``

But my question remains...maybe we should just set ``journald`` to be the default instead?  It's great at managing logs :smile:

Reply:
If setting the log-driver to `journald` as default fixes all these issues, I agree and vote to do that instead!

Reply:
>Edit: The user override case is handled in the (first) if branch above.

Ah, I see. I'll run a few tests to verify that.; Sorry for the noise

>maybe we should just set journald to be the default instead?

I think its a good idea, last time we had an issue with rate limiting - this would skip some log entries. Not sure if we configure journald properly now